src/test/kotlin/uk/gov/dluhc/printapi/rest/GenerateTemporaryCertificateIntegrationTest.kt

```diff
@@ -133,4 +152,125 @@ internal class GenerateTemporaryCertificateIntegrationTest : IntegrationTest() {
+            .bearerToken(getBearerToken(eroId = ERO_ID, groups = listOf("ero-$ERO_ID", "ero-vc-admin-$ERO_ID")))
+            .contentType(MediaType.APPLICATION_JSON)
+            .withBody(requestBody)
+            .exchange()
+            .expectStatus()
+            .isBadRequest
+            .returnResult(ErrorResponse::class.java)
+
+        // Then
+        val actual = response.responseBody.blockFirst()
+        assertThat(actual)
+            .hasStatus(400)
+            .hasError("Bad Request")
+            .hasMessageContaining("Temporary Certificate gssCode '$GSS_CODE' does not exist")
+    }
+
+    @Test
+    fun `should return not found given no local authority found for the provided ero and gss code`() {
+        // Given
+        val eroResponseHasDifferentEroIdToBearerTokenAndUri = buildElectoralRegistrationOfficeResponse(
+            id = OTHER_ERO_ID,
+            localAuthorities = listOf(buildLocalAuthorityResponse(gssCode = GSS_CODE))
+        )
+        wireMockService.stubCognitoJwtIssuerResponse()
+        wireMockService.stubEroManagementGetEroByGssCode(eroResponseHasDifferentEroIdToBearerTokenAndUri, GSS_CODE)
+        val requestBody = buildGenerateTemporaryCertificateRequest(gssCode = GSS_CODE)
+
+        // When
+        val response = webTestClient.post()
+            .uri(URI_TEMPLATE, ERO_ID)
+            .bearerToken(getBearerToken(eroId = ERO_ID, groups = listOf("ero-$ERO_ID", "ero-vc-admin-$ERO_ID")))
+            .contentType(MediaType.APPLICATION_JSON)
+            .withBody(requestBody)
+            .exchange()
+            .expectStatus()
+            .isBadRequest
+            .returnResult(ErrorResponse::class.java)
+
+        // Then
+        val actual = response.responseBody.blockFirst()
+        assertThat(actual)
+            .hasStatus(400)
+            .hasError("Bad Request")
+            .hasMessageContaining("Temporary Certificate gssCode '$GSS_CODE' is not valid for eroId '$ERO_ID'")
+    }
+
+    @Test
+    fun `should return temporary certificate pdf given valid request for authorised user`() {
+        // Given
+        val photoLocationArn = addPhotoToS3()
+        val request = buildGenerateTemporaryCertificateRequest(photoLocation = photoLocationArn)
+        val localAuthorityName = aValidLocalAuthorityName()
+        val localAuthorities: List<LocalAuthorityResponse> = listOf(
+            buildLocalAuthorityResponse(
+                gssCode = request.gssCode,
+                contactDetailsEnglish = buildContactDetails(name = localAuthorityName)
+            )
+        )
+        val eroResponse = buildElectoralRegistrationOfficeResponse(id = ERO_ID, localAuthorities = localAuthorities)
+        wireMockService.stubCognitoJwtIssuerResponse()
+        wireMockService.stubEroManagementGetEroByGssCode(eroResponse, request.gssCode)
+
+        // When
+        val response = webTestClient.mutate()
+            .codecs { it.defaultCodecs().maxInMemorySize(MAX_SIZE_1_MB) }
```

Comment:
I've done this differently in my [PR](https://github.com/cabinetoffice/eip-ero-print-api/pull/140)

Reply:
Yes - I didn't like setting the property in the application context as kind of hides where it's needed.  Wanted to be clear it was only issue with the test client and not the controller code.